### ISSUE TITLE
Error importing decoupler library in python

### ISSUE THREAD

**Issue by aimorenov-owkin** [saezlab/decoupler-py#55]
Hello, I installed the decoupler library (v 1.4.0) in a python environment : python v 3.9; numpy v 1.24.3

Installation was successful but when I try to import the library (`import decoupler as dc`) I get the error: `AttributeError: module 'numpy' has no attribute 'long'`

I think that this is an issue with the version of numpy from what I found on this post: https://levelup.gitconnected.com/fix-attributeerror-module-numpy-has-no-attribute-float-d7d68c5a4971. 

Would there be an update of the numpy code found in decoupler? I need to use a version of numpy >1.2 so downgrading numpy would not be the best solution for me. 

Thank you!

**Comment by PauBadiaM** [comment]
Hi @aimorenov-owkin 

Version `1.23.5` should be fine. Else, could you paste the complete error? This way I can pinpoint exactly where it breaks, thanks!

**Comment by aimorenov-owkin** [comment]
Thank you for your answer @PauBadiaM. The detailed error that I get is the following : 
**Initial error in python 3.9, numpy 1.24.3**
```
/home/owkin/conda_envs/tde-amorenovega/lib/python3.9/site-packages/numba/types/__init__.py:110: FutureWarning: In the future `np.long` will be defined as the corresponding NumPy scalar.
  long_ = _make_signed(np.long)
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[2], line 1
----> 1 import decoupler as dc

File ~/conda_envs/tde-amorenovega/lib/python3.9/site-packages/decoupler/__init__.py:9
      7 from .utils_anndata import get_acts, swap_layer, get_pseudobulk, get_contrast, rank_sources_groups  # noqa: F401
      8 from .utils_anndata import get_top_targets, format_contrast_results, filter_by_expr, filter_by_prop  # noqa: F401
----> 9 from .method_wmean import run_wmean  # noqa: F401
     10 from .method_wsum import run_wsum  # noqa: F401
     11 from .method_ulm import run_ulm  # noqa: F401

File ~/conda_envs/tde-amorenovega/lib/python3.9/site-packages/decoupler/method_wmean.py:11
      8 from scipy.sparse import csr_matrix
     10 from .pre import extract, match, rename_net, get_net_mat, filt_min_n
---> 11 from .method_gsea import std
     13 from anndata import AnnData
     14 from tqdm import tqdm

File ~/conda_envs/tde-amorenovega/lib/python3.9/site-packages/decoupler/method_gsea.py:18
     15 from anndata import AnnData
     16 from tqdm import tqdm
---> 18 import numba as nb
     21 @nb.njit(nb.types.Tuple((nb.f4, nb.i8, nb.f4[:]))(nb.f4[:], nb.i8[:], nb.b1[:], nb.f4), cache=True)
     22 def compute_es_per_rank(row, rnks, set_msk, dec):
     23 
     24     # Init empty
     25     mx_value = 0.0

File ~/conda_envs/tde-amorenovega/lib/python3.9/site-packages/numba/__init__.py:15
     12 __version__ = get_versions()['version']
     13 del get_versions
---> 15 from . import config, errors, _runtests as runtests, types
     17 # Re-export typeof
     18 from .special import (
     19     typeof, prange, pndindex, gdb, gdb_breakpoint, gdb_init,
     20     literally, literal_unroll
     21 )

File ~/conda_envs/tde-amorenovega/lib/python3.9/site-packages/numba/types/__init__.py:110
    108 intc = _make_signed(np.intc) # C-compat int
    109 uintc = _make_unsigned(np.uintc) # C-compat uint
--> 110 long_ = _make_signed(np.long)
    111 ulong = _make_unsigned(np.long)
    112 longlong = _make_signed(np.longlong)

File ~/conda_envs/tde-amorenovega/lib/python3.9/site-packages/numpy/__init__.py:320, in __getattr__(attr)
    317     from .testing import Tester
    318     return Tester
--> 320 raise AttributeError("module {!r} has no attribute "
    321                      "{!r}".format(__name__, attr))

AttributeError: module 'numpy' has no attribute 'long'
```

**New error in python 3.9, numpy 1.23.5**
When I downgrade numpy to version  1.23.5 as you suggested (python 3.9), I get the following error (see attached txt file for details): 
````
Type Error: create_target_machine got an unexpected keyword argument 'jitdebug'

````

Following the following [post](github.com/rapidsai/cuml/issues/2389) on such error, I tried downgrading llvmlite and upgrading numba (llvmlite downgrade from 0.39.1 -> 0.32.1; numba update from 0.47.0 -> 0.49.1). However this is only possible in a python <3.9 so had to recreate an environment with python 3.8 and under this setting it works. 

Tested also under python 3.10 and it works (just need to correct maptplotlib version). 

So for python v3.9 on my side it does not appear to work.

**Comment by PauBadiaM** [comment]
Hi @aimorenov-owkin 

Your error shows that there is an incompatibility between `numpy` and `numba`. I just tried installing decoupler in a fresh `python=3.9` environment with `numpy==1.24.3` and it works fine. Maybe update `numba` to `numba==0.57.1`:
```
conda create -n=test python=3.9
conda activate test

pip install numpy==1.24.3
pip install decoupler==1.4.0
```

**Comment by aimorenov-owkin** [comment]
Dear @PauBadiaM ,

Thank you very much for looking further into this. I checked and indeed the test environment you created with `numpy==1.24.3` and `python =3.9` works.

I found the reason why it was not working on my side and it is because in the repository that I am working on the llvmlite version was fixed, meaning I was getting errors with numba which could not be updated due to this. 

Thanks again,

Aura
